q2_moshpit/busco/plots_summary.py

```diff
@@ -0,0 +1,139 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2023, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+from typing import List
+
+import altair as alt
+import pandas as pd
+
+alt.data_transformers.disable_max_rows()
+
+LABEL_FONT_SIZE = 12
+TITLE_FONT_SIZE = 15
+PLOT_DIM = 260
+
+
+def _draw_horizontal_histograms(data: pd.DataFrame, columns: List[str]):
+    data = pd.melt(
+        data,
+        id_vars=["sample_id", "mag_id", "dataset", "n_markers"],
+        value_vars=columns,
+        value_name="metric",
+        var_name="category",
+    )
+
+    charts = []
+    for i, category in enumerate(columns):
+        x_title = category.replace("_", " ").capitalize()
+        y_title = "MAG count" if i == 0 else None
+        chart = alt.Chart(
+            data[data["category"] == category]
+        ).mark_bar().encode(
+            x=alt.X('metric:Q', bin=True, title=x_title),
+            y=alt.Y('count()', title=y_title),
+            color=alt.value("steelblue")
+        ).properties(
+            width=PLOT_DIM,
+            height=PLOT_DIM
+        )
+        charts.append(chart)
+
+    chart = alt.hconcat(
+        *charts
+    )
+
+    return chart
+
+
+def _draw_marker_summary_histograms(data: pd.DataFrame) -> dict:
+    """
+    Draws summary histograms for the BUSCO marker results of all samples.
+
+    Returns:
+        dict: Dictionary containing the Vega spec.
+    """
+    chart = _draw_horizontal_histograms(
+        data, columns=["single", "duplicated", "fragmented", "missing"]
+    )
```

Comment:
![Screenshot 2024-04-25 at 10 45 18](https://github.com/bokulich-lab/q2-moshpit/assets/54123712/a819d563-14d0-499f-8413-ea4e174002f1)

I think it would be useful if the domain and the range of the top-row histograms was the same for all four categories (i.e. 0-100 domain and 0-'max_counts_in_any_bin' range). This might make the top-row histogram comparison more informative. I think one can do this easily in Altair, either by [making a faceted histogram](https://altair-viz.github.io/gallery/distributions_faceted_histogram.html) or by [manually setting the scale of each chart](https://altair-viz.github.io/user_guide/customization.html#adjusting-axis-labels:~:text=To%20specify%20exact,Horsepower%3AQ%27%0A)).

Reply:
Actually, the way it is shown now is intentional, after a few rounds of testing different options. The reason is that for cases where you have a plot with fewer bins than in the other plots, this subplot would become pretty much unreadable. That's why I opted for the autoscaling - this way we're sure that one can always read what's on the plot.

Reply:
I see. That makes sense 🙃